libsolidity/parsing/Parser.cpp

```diff
@@ -823,13 +823,22 @@ ASTPointer<ParameterList> Parser::parseParameterList(
 	VarDeclParserOptions options(_options);
 	options.allowEmptyName = true;
 	expectToken(Token::LParen);
+
 	if (!_allowEmpty || m_scanner->currentToken() != Token::RParen)
 	{
 		parameters.push_back(parseVariableDeclaration(options));
+
+		if(m_scanner->currentToken() == Token::Comma && m_scanner->peekNextToken() == Token::RParen){
+			fatalParserError("Unexpected trailing comma in arguments");
+		}
+
 		while (m_scanner->currentToken() != Token::RParen)
 		{
 			expectToken(Token::Comma);
```

Comment:
How about replacing those two checks with a single one before this `expectToken`? I think that should work.

Reply:
Also please follow the coding style (space after if, before {, also no brackets needed if the body of a conditional is a single expression).